### PR TITLE
[MIG] migrate module 'project_billing_utils' to v8

### PR DESCRIPTION
**Pull request by damdam-s**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please extract module description to a README.rst file

**Comment by damdam-s** [comment]
update :)

File: project_billing_utils/invoice.py [hunk 1]
```diff
@@ -18,25 +18,22 @@
...
-        if 'special_search' not in context:
-            return super(AccountInvoice, self).name_get(
-                cr, uid, ids, context=context)
+    @api.multi
+    def name_get(self):
+        if 'special_search' not in self.env.context:
+            return super(AccountInvoice, self).name_get()
         else:
-            if not ids:
+            if not self.ids:
```
Comment (gurneyalex) [inline project_billing_utils/invoice.py]: `if not self:`

(`Model` defines `__nonzero__`)

File: project_billing_utils/invoice.py [hunk 2]
```diff
@@ -44,29 +41,22 @@ def name_get(self, cr, uid, ids, context=None):
                 # We will
```
Comment (gurneyalex) [inline project_billing_utils/invoice.py:46]: you don't need  `read`. Something like the following should do the same (just check the bit with partner_id)

```
for rec in self:
    rest.append((rec.id, ('%s - %s - %s' % (rec.number or '',
                                     rec.partner_id.name_get(), rec.name or ''))))
```

File: project_billing_utils/project.py [hunk 3]
```diff
@@ -18,31 +18,27 @@
...
+        account_line_obj = self.env['account.analytic.line']
+        for project in self:
+            account_lines = account_line_obj.search(
+                [('account_id', '=', project.analytic_account_id.id)])
             # If we found line linked with account we raise an error
-            if account_line_ids:
-                raise osv.except_osv(
+            if account_lines:
+                raise exceptions.Warning(
                     _('Invalid Action !'),
```
Comment (gurneyalex) [inline project_billing_utils/project.py]: please remove the exclamation mark (and update the translation files)

File: project_billing_utils/project.py [hunk 4]
```diff
@@ -18,31 +18,27 @@
...
                     _('You can\'t delete account %s , analytic lines linked '
                       'to it' % project.name))
```
Comment (gurneyalex) [inline project_billing_utils/project.py]: should also be fixed in 7.0: the call to _() should not feature the interpolated string: move  `% project _name` out of the call

bonus point if you change the sentence to proper english :smile_cat:

File: project_billing_utils/wizard/blank_invoice.py [hunk 5]
```diff
@@ -18,45 +18,42 @@
...
         """
         Prepare the values used for the invoice creation.
         Override and properly use the `super` chain
         to customize the invoice creation
         """
         if not project.partner_id:
-            raise osv.except_osv(
+            raise exceptions.Warning(
                 _('UserError'),
                 _('The Partner is missing on the project:\n%s' % project.name))
```
Comment (gurneyalex) [inline project_billing_utils/wizard/blank_invoice.py]: same remark as above about `_(string % value)` -> `_(string) % value`

File: project_billing_utils/wizard/blank_invoice.py [hunk 6]
```diff
@@ -18,45 +18,42 @@
...
 
         if not project.pricelist_id:
-            raise osv.except_osv(
+            raise exceptions.Warning(
                 _('UserError'),
                 _('The Customer Pricelist is '
                   'missing on the project:\n%s' % project.name))
```
Comment (gurneyalex) [inline project_billing_utils/wizard/blank_invoice.py]: and again

File: project_billing_utils/wizard/open_invoices.py [hunk 7]
```diff
@@ -18,47 +18,48 @@
...
         else:
             xml_id = 'action_invoice_tree2'
-        result = mod_obj.get_object_reference(cr, uid, 'account', xml_id)
-        id = result and result[1] or False
-        result = act_obj.read(cr, uid, id, context=context)
-        invoice_domain = eval(result['domain'])
+
+        result = self.env.ref('account.%s' % xml_id)
+        if result:
+            result = result.read()[0]
```
Comment (gurneyalex) [inline project_billing_utils/wizard/open_invoices.py]: result is a browse record (I think). Why call `read()` ?

**Comment by gurneyalex** [comment]
:+1:

Can you make a PR on the 7.0 branch for the issues you fixed here?

**Comment by damdam-s** [comment]
@gurneyalex : fix issues in v7 : https://github.com/OCA/project-reporting/pull/7

**Comment by pedrobaeza** [comment]
:+1:

**Comment by lepistone** [comment]
thanks

**Pull request merged**